src/game_map.cpp

```diff
@@ -918,25 +935,61 @@ void Game_Map::UpdateEncounterSteps() {
 		return;
 	}
 
-	location.encounter_steps = location.encounter_steps - terrain->encounter_rate;
```

Comment:
the reason this was written in such an ugly way (not using -=) was btw for support in my automagical refactor liblcf field-access to Get/Set function converter